### PR TITLE
Rework maintainer documentation

### PR DESCRIPTION
**Pull request by greg0ire**
I'm looking for a place to write things about label standardization, and I think this page would be a great fit. It is outdated though, so let's overhaul it.

Tips about setting up repositories are removed because being a
proficient contributor is kind of a requirement, and new maintainers
probably do not need to be taught how to setup a git
repository, or how to add remotes.
References to master, or to the merge down workflow are dropped as well.

### CODE REVIEW

**Comment by greg0ire** [comment]
@morozov @derrabus I'm not sure why I can't request reviews from you, but please review as well

**Comment by morozov** [comment]
Probably because we're not approvers for this repo.

Do people really use `git switch --create` instead of `git checkout -b`? It will take me another decade to adjust. J/k, looks good.

File: source/contribute/maintainer.rst [hunk 1]
```diff
@@ -10,129 +10,53 @@ access to the main repository of a project. In the example of the ORM,
...
+- Branch naming convention: 1.0.x, 2.0.x, 2.1.x
 
-Release branches are created when **master** has reached the state of
-the next major or minor release. They allow for continuous bug fixes and
-patch releases of that particular release until the release is no longer
-supported.
+Release branches are created either manually or through
+``laminas/automatic-releases`` (depending on the configuration) when the
+next minor or major branch has reached the state of the next major or
+minor release, so that contributors can submit new features.
```
Comment (SenseException) [inline source/contribute/maintainer.rst:46]: I don't understand that sentence. It sounds like an "or" or an "and" is missing.

Comment (derrabus) [reply source/contribute/maintainer.rst:46]: I also think that this sentence should be split into two or more sentences to make it easier to understand.

Comment (greg0ire) [reply source/contribute/maintainer.rst:46]: I don't think any word is missing but it's indeed hard to read. I'll rework it.

**Review by SenseException — APPROVED** [review]

**Review by derrabus — APPROVED** [review]

**Pull request merged**